### PR TITLE
feat(pages): add EOY dialog to home page

### LINKED ISSUE #237: Do EOY pop up on Free.law
A modal pop-up like the following should be placed onto Free.law starting Thursday or so:

It should have an X in the upper right that dismisses it for one week.

This version should run from 12/4 (or so) to 12/20:

Image

Link to:  https://donate.free.law/forms/supportflp

This version should run from (about) 12/21 to 12/31:

Image

Both should link to: https://donate.free.law/forms/supportflp

The final version doesn't have to look as good at @jeniferw's mock ups this year, since we're short on time (my fault!), but we have assets here if they're helpful: 

https://drive.google.com/drive/folders/1jqeM6NVYWQ0sQTRdI5zm1TEVklSKRAJD

First step, can you figure out the tailwind library we're using for UI components and find their modal options, please, @elisa-a-v, and then from there, we can start putting our content into it?

### PR DESCRIPTION
**Pull request by elisa-a-v**
Closes #237

## Changes
- Added a reusable and generic `Dialog` component which can be managed independently, or externally by providing props.
- Added `EOYDialog` component that:
  - Displays two versions of the dialog in specified date ranges.
  - Can be dismissed for a week
    - Each version is independendently dismissed
    - Dismissal states are stored separately for each version using `localStorage`.
  - Can be enabled/disabled using `EOY_MODE` const at the top of the `/components/dialog.js` file.
  - Its contents can be easily modified by changing the constants at the top of the file:
    - End dates for each version: `FIRST_EOY_DIALOG_END_MONTH`, `FIRST_EOY_DIALOG_END_DAY`, `SECOND_EOY_DIALOG_END_MONTH`, `SECOND_EOY_DIALOG_END_DAY`
    - Text copies for each version: `FIRST_EOY_TITLE`, `SECOND_EOY_TITLE`, `FIRST_EOY_TEXT`, `SECOND_EOY_TEXT`

---
# Screenshots
#### Large screens
image

#### Medium screens
image

#### Small screens
image

#### Large screens
image

#### Medium screens
image

#### Small screens
image

#### To do:
- [x] Dismiss dialog for a week
- [x] Add second dialog
- [x] Set schedule for displaying each dialog
- [x] Add setting to handle dialog visibility from env variables

### CODE REVIEW

**Comment by elisa-a-v** [comment]
@mlissner since free.law doesn't have env variables at all, I thought it was probably good enough to just add a constant at the top of the file, along with other constants that could be modified from year to year. What do you think?

File: components/dialog.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+ *
+ * @returns {JSX.Element|null} The EOYDialog component or null if not displayed.
+ */
+export function EOYDialog() {
+  if (!EOY_MODE) return;
+
+  // DIALOG CLASSES
+  const dialogBaseClasses =
+    'bg-center bg-cover bg-no-repeat flex flex-col justify-between' +
+    ' items-center max-w-screen-md w-full min-h-450px max-h-screen border-purple-800';
```
Comment (mlissner) [inline components/dialog.js:49]: This seems about right to me, thank you, but let's remove the purple border. I think simplicity is better on that.

File: components/dialog.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+}) {
+  const dialogId = useId();
+  const [isOpenLocal, setIsOpenLocal] = useState(true);
+  let isOpen, setIsOpen;
+  if (isOpenProp == null || setIsOpenProp == null) {
+    [isOpen, setIsOpen] = [isOpenLocal, setIsOpenLocal];
+  } else {
+    [isOpen, setIsOpen] = [isOpenProp, setIsOpenProp];
+  }
+  let classes = 'fixed p-6 bg-purple-900 border-4';
```
Comment (mlissner) [inline components/dialog.js:183]: I think we can remove the border thing here too, right?

**Review by mlissner — APPROVED** [review]
Looks good. Impressive work getting it to match Jenifer's design. One little aesthetic tweak from me, and I'll defer to Eduardo for a final review of the code.

**Comment by mlissner** [comment]
@jeniferw, if you want to see the modal for free.law, check out the deploy preview above.

**Comment by elisa-a-v** [comment]
@jeniferw I originally made the title and button text bolder like the mockups, but at some stage it stopped working so I just left it like that. I now figured out how to fix it and it's extremely easy to do. I think thinner looks slightly more aligned with the rest of the site's styles to me, while bolder is easier to read. I can change it no problem so let me know which one you prefer.

image
image

**Review by ERosendo — APPROVED** [review]
Looks good!! Thanks @elisa-a-v

**Comment by mlissner** [comment]
Looks great, so I went ahead and merged!

**Pull request merged**